clouddriver-ecs/src/test/groovy/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperationSpec.groovy

```diff
@@ -1234,4 +1261,107 @@ class CreateServerGroupAtomicOperationSpec extends CommonAtomicOperation {
       "test-cluster"
     )
   }
+
+  def 'should create no tags by default'() {
+    given:
+    def description = Mock(CreateServerGroupDescription)
+
+    description.getApplication() >> 'mygreatapp'
+    description.getStack() >> 'stack1'
+    description.getFreeFormDetails() >> 'details2'
+    description.getTags() >> null
+
+    def operation = new CreateServerGroupAtomicOperation(description)
+
+    when:
+    def request = operation.makeServiceRequest("taskDefArn", new EcsServerGroupName('mygreatapp-stack1-details2-v011'), 1,  NamerRegistry.getDefaultNamer(), false)
+
+    then:
+    assert request.enableECSManagedTags == null
+    assert request.propagateTags == null
+    def tags = request.getTags()
+    assert tags.isEmpty()
+  }
+
+  def 'should create moniker tags if enabled and no other tags'() {
+    given:
+    def description = Mock(CreateServerGroupDescription)
+
+    description.getApplication() >> 'mygreatapp'
+    description.getStack() >> 'stack1'
+    description.getFreeFormDetails() >> 'details2'
+    description.getTags() >> null
+
+    def operation = new CreateServerGroupAtomicOperation(description)
+
+    when:
+    def request = operation.makeServiceRequest("taskDefArn", new EcsServerGroupName('mygreatapp-stack1-details2-v011'), 1, new EcsTagNamer(), true)
+
+    then:
+    assert request.enableECSManagedTags == true
+    assert request.propagateTags == PropagateTags.SERVICE.toString()
+    def tags = request.getTags()
+    assert tags.size() == 5
+    tags.contains(new Tag(key: EcsTagNamer.APPLICATION, value: 'mygreatapp'))
+    tags.contains(new Tag(key: EcsTagNamer.CLUSTER, value: 'mygreatapp-stack1-details2'))
+    tags.contains(new Tag(key: EcsTagNamer.STACK, value: 'stack1'))
+    tags.contains(new Tag(key: EcsTagNamer.DETAIL, value: 'details2'))
+    tags.contains(new Tag(key: EcsTagNamer.SEQUENCE, value: '11'))
+  }
+
+  def 'should create custom tags if moniker not enabled'() {
+    given:
+    def description = Mock(CreateServerGroupDescription)
+
+    description.getApplication() >> 'mygreatapp'
+    description.getStack() >> 'stack1'
+    description.getFreeFormDetails() >> 'details2'
+    description.getTags() >> ['label1': 'value1', 'fruit':'tomato']
+
+    def operation = new CreateServerGroupAtomicOperation(description)
+
+    when:
+    def request = operation.makeServiceRequest("taskDefArn", new EcsServerGroupName('mygreatapp-stack1-details2-v011'), 1,  NamerRegistry.getDefaultNamer(), true)
+
+    then:
+    assert request.enableECSManagedTags == true
+    assert request.propagateTags == 'SERVICE'
+    def tags = request.getTags()
+    assert tags.size() == 2
+    tags.contains(new Tag(key: 'label1', value: 'value1'))
+    tags.contains(new Tag(key: 'fruit', value: 'tomato'))
+  }
+
+  def 'should create custom tags and moniker tags if moniker enabled'() {
```

Comment:
Can you also add a test for the exception case - when moniker support is enabled (enableTags) but the account is not configured to use the long arn format?